core/dbt/exceptions.py

```diff
@@ -798,6 +808,17 @@ def raise_duplicate_macro_patch_name(patch_1, patch_2):
     )
 
 
+def raise_duplicate_source_patch_name(patch_1, patch_2):
+    name = f'{patch_1.source_name}.{patch_1.table_name}'
+    raise_compiler_error(
```

Comment:
I saw an unhandled exception here while testing:
```
  File "/Users/drew/fishtown/dbt/core/dbt/task/runnable.py", line 79, in _runtime_initialize
    self.load_manifest()
  File "/Users/drew/fishtown/dbt/core/dbt/task/runnable.py", line 67, in load_manifest
    self.manifest = get_full_manifest(self.config)
  File "/Users/drew/fishtown/dbt/core/dbt/perf_utils.py", line 23, in get_full_manifest
    return load_manifest(config, internal, set_header)
  File "/Users/drew/fishtown/dbt/core/dbt/parser/manifest.py", line 634, in load_manifest
    return ManifestLoader.load_all(config, internal_manifest, macro_hook)
  File "/Users/drew/fishtown/dbt/core/dbt/parser/manifest.py", line 355, in load_all
    loader.load(internal_manifest=internal_manifest)
  File "/Users/drew/fishtown/dbt/core/dbt/parser/manifest.py", line 211, in load
    self.parse_project(project, macro_manifest, old_results)
  File "/Users/drew/fishtown/dbt/core/dbt/parser/manifest.py", line 185, in parse_project
    self.parse_with_cache(path, parser, old_results)
  File "/Users/drew/fishtown/dbt/core/dbt/parser/manifest.py", line 141, in parse_with_cache
    parser.parse_file(block)
  File "/Users/drew/fishtown/dbt/core/dbt/parser/schemas.py", line 526, in parse_file
    for test_block in parser.parse():
  File "/Users/drew/fishtown/dbt/core/dbt/parser/schemas.py", line 612, in parse
    self.results.add_source_patch(self.yaml.file, patch)
  File "/Users/drew/fishtown/dbt/core/dbt/parser/results.py", line 173, in add_source_patch
    raise_duplicate_source_patch_name(patch, self.source_patches[key])
  File "/Users/drew/fishtown/dbt/core/dbt/exceptions.py", line 812, in raise_duplicate_source_patch_name
    name = f'{patch_1.source_name}.{patch_1.table_name}'
AttributeError: 'SourcePatch' object has no attribute 'source_name'
```

I got here by adding two source patches for the same source: eg,
```
models/source1.yml
models/source2.yml
dbt_modules/package/models/source.yml
```

`source1.yml` and `source2.yml` both patch the `source.yml` file from the dependency package

Reply:
I suppose this error message should say:

```
dbt found two schema.yml entries which override the source '{name}'.
Sources may only be overridden once in a project. To fix this,
remove the source override for {name} in one of these files....
```

Or something like that. FWIW I don't think these patches currently contain file paths? Would be good to include those too!

Reply:
Yeah, this is an oversight, thanks for catching it: source patches should definitely have file paths (and should probably set an `Optional` `ParsedSourceDefinition.patch_path`). I'll add a test for this, too.